File: Orquestra.Domain/Entities/ServiceOrder.cs
using Orquestra.Domain.Enums;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Orquestra.Domain.Entities;

public sealed class ServiceOrder : Audit
{
    [Key]
    public Guid ServiceOrderId { get; set; }

    public Guid CompanyId { get; set; }
    [ForeignKey(nameof(CompanyId))]
    public Company? Company { get; set; }

    public Guid? QuoteId { get; set; } // Nullable;
    [ForeignKey(nameof(QuoteId))]
    public Quote? Quote { get; set; }

    public Guid ClientId { get; set; }
    [ForeignKey(nameof(ClientId))]
    public Client? Client { get; set; }

    [MaxLength(120)]
    public string? Title { get; set; }

    [MaxLength(255)]
    public string? Observation { get; set; }

    public DateTime? ExecutionDate { get; set; }

    public ServiceOrderStatusEnum ServiceOrderStatus { get; set; } = ServiceOrderStatusEnum.Pending;
}

File: Orquestra.Frontend/app/api/consts/service-order.ts
import { BASE } from '@/app/api/fetch';
import { Guid } from 'guid-typescript';
import { iClient } from './client';
import { iCompanyOutput } from './company';
import { iQuote } from './quote';

const controller = 'api/ServiceOrder';

export const CONSTS_SERVICE_ORDER = {
    post: `${BASE}/${controller}`,
    put: `${BASE}/${controller}`,
    get: `${BASE}/${controller}`,
    getAllByCompanyId: `${BASE}/${controller}/GetAllByCompanyId`,
    disable: `${BASE}/${controller}/Disable`
};

export interface iServiceOrder {
    serviceOrderId?: Guid;
    companyId?: Guid;
    company?: iCompanyOutput | null;

    quoteId?: Guid | null;
    quote?: iQuote | null;

    clientId?: Guid;
    client?: iClient | null;

    title?: string | null;
    observation?: string | null;

    executionDate?: Date | string | null;

    serviceOrderStatus?: string | null;

    createdDate?: Date | null;
}

export interface iServiceOrderPaginated {
    output: iServiceOrder[];
    count: number;
}